### ISSUE TITLE
Setting customer.source = token adds the source, meaning customer.sources.first will be the old card.

### ISSUE THREAD

**Issue by TrevorHinesley** [rebelidealist/stripe-ruby-mock#194]
`customer.source = token` should act the same as `customer.card = token`, but instead it's acting the same as adding a card to a customer, rather than replacing the default.

**Comment by TrevorHinesley** [comment]
May have spoken too soon, something's up but this may not be the root problem.

**Comment by TrevorHinesley** [comment]
So the issue is the default is set, but Stripe's API gets rid of the old source and adds the new one to the list, as well as setting it as the default. Currently, the mock gem adds it to sources, so there are two cards, but sets the default_source. This means any legacy code in someone's app that uses `customer.cards.first` (which is very common) will fail, because `.first` is the old card, the last card is the newest one. Also, #170 is still present in the update.

**Comment by TrevorHinesley** [comment]
I don't have the time in the next couple days to add this in myself, but for anyone looking at this, I believe the issue is here:

```ruby
      def add_card_to_object(type, card, object, replace_current=false)
        card[type] = object[:id]
        cards_or_sources = object[:cards] || object[:sources]

        is_customer = object.has_key?(:sources)

        if replace_current
          cards_or_sources[:data].delete_if {|card| card[:id] == object[:default_card]}
          object[:default_card]   = card[:id] unless is_customer
          object[:default_source] = card[:id] if is_customer
        else
          cards_or_sources[:total_count] += 1
        end

        object[:default_card]   = card[:id] if !is_customer && object[:default_card].nil?
        object[:default_source] = card[:id] if is_customer  && object[:default_source].nil?
        cards_or_sources[:data] << card

        card
      end
```

And I think the fix is as easy as this (untested):

```ruby
      def add_card_to_object(type, card, object, replace_current=false)
        card[type] = object[:id]
        cards_or_sources = object[:cards] || object[:sources]

        is_customer = object.has_key?(:sources)

        if replace_current
          cards_or_sources[:data].delete_if {|card| card[:id] == object[:default_card]}
          object[:default_card]   = card[:id] unless is_customer
          object[:default_source] = card[:id] if is_customer
          cards_or_sources[:data] = [card]
        else
          cards_or_sources[:total_count] += 1
          cards_or_sources[:data] << card
        end

        object[:default_card]   = card[:id] if !is_customer && object[:default_card].nil?
        object[:default_source] = card[:id] if is_customer  && object[:default_source].nil?

        card
      end
```

**Comment by TrevorHinesley** [comment]
Fixed in #195

**Comment by TrevorHinesley** [comment]
Is anyone checking pull requests right now? Sorry to keep badgering, this is just an important testable feature.
